src/main/java/org/pesmypetcare/webservice/controller/petmanager/PetMedicationRestController.java

```diff
@@ -0,0 +1,159 @@
+package org.pesmypetcare.webservice.controller.petmanager;
+
+import org.pesmypetcare.webservice.entity.MedicationEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+import org.pesmypetcare.webservice.service.MedicationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@RestController
+@RequestMapping("/medication")
+public class PetMedicationRestController {
+    @Autowired
+    private MedicationService medicationService;
+    private final String TOKEN = "token";
+
+    /**
+     * Creates a Medication eaten by a pet on the database.
+     * @param accessToken oauth2 token needed to access the Database
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName Date + name of the Medication
+     * @param medication The Medication entity that contains the attributes of the Medication
+     */
+    @PostMapping("/{owner}/{petName}/{dateName}")
+    public void createMedication(@RequestHeader(TOKEN) String accessToken, @PathVariable String owner,
+                                 @PathVariable String petName,
+                                 @PathVariable String dateName,
+                                 @RequestBody MedicationEntity medication) {
+        medicationService.createMedication(owner, petName, dateName, medication);
+    }
+
+    /**
+     * Deletes all the medication with the specified owner and pet from certain date.
+     * @param accessToken oauth2 token needed to access the Database
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName date + name of the receipt of the medication
+     */
+    @DeleteMapping("/{owner}/{petName}/{dateName}")
+    public void deleteByDateAndName(@RequestHeader(TOKEN) String accessToken,
+                                    @PathVariable String owner,
+                                    @PathVariable String petName,
+                                    @PathVariable  String dateName) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a single space to separate non-whitespace characters.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)